src/hoconsc.erl

```diff
@@ -48,10 +49,22 @@ union(OfTypes) when is_list(OfTypes) -> ?UNION(OfTypes).
 %% @doc make a enum type.
 enum(OfSymbols) when is_list(OfSymbols) -> ?ENUM(OfSymbols).
 
+%% @doc make a lazy type.
 lazy(HintType) -> ?LAZY(HintType).
 
+%% @doc make a map type.
 map(Name, Type) -> ?MAP(Name, Type).
 
+%% @doc Check Type is a hocon type.
+is_type(?UNION(Members)) -> lists:all(fun is_type/1, Members);
+is_type(?ARRAY(ElemT)) -> is_type(ElemT);
+is_type(?LAZY(HintT)) -> is_type(HintT);
+is_type(?REF(_)) -> true;
+is_type(?R_REF(_, _)) -> true;
+is_type(#{type := _}) -> true;
```

Comment:
schema includes type.
schema can be a map or an anonymous function.
From a schema one can retrieve either type or other metadata

maybe name it `is_schema` instead

Reply:
done @zmstone